src/community-toolbox.js

```diff
@@ -39,6 +39,130 @@ CommunityToolbox = function CommunityToolbox(org, repo) {
        .then(callback);
   }
 
+  function fetchRepoContributors(org, repo) {
+
+    var arr = [];
+    api.Repositories
+           .getRepoContributors(org, repo, {method: "HEAD", qs: { sort: 'pushed', direction: 'desc', per_page: 100 } })
+           .then(function(contribData) {
+             var headers = contribData;
+             if (headers.hasOwnProperty("link")) {
+                 var parsed = parse(headers['link']);
+                 totalPages = parseInt(parsed.last.page);
+             } else {
+                 totalPages = 1;
+             }
+             return totalPages;
+           })
+           .then((totalPages) => {
+              for(var i = 1; i <= totalPages; i++) {
+                api.Repositories
+                .getRepoContributors(org, repo, { method:"GET", qs: { sort: 'pushed', direction: 'desc', per_page: 100, page:i } })
+                .then(function(contributors) {
+                    if (contributors != null || contributors.length > 0) {
+                      contributors.map((contributor, i) => arr.push(contributor));
+                    }
+                });
+              }
+           })
+
+
+  return new Promise((resolve, reject) => {
+    resolve(arr);
+  })
+}
+
+
+
+  // Fetches all the publiclab repos
+  function getAllRepos(org) {
+    let repos = [];
+    fetch(`https://api.github.com/orgs/${org}/repos`, {'headers': { "Content-Type": "application/x-www-form-urlencoded", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36" }})
+    .then((response) => {
+      if (response.status == "200") {
+        return response.json();
+      }
+    })
+    .then((res) => {
+      res.map((item, i) => {
+        repos[i] = item.name;
+      });
+      // Stores all public lab repos to localstorage (only first 30 repos yet)
+      localStorage.setItem('repos', JSON.stringify(repos));
+    })
+  }
+
+  // Fetches all the UNIQUE contributors from all the publiclab repos
+  // and stores them in localstorage in the form of array
+  function getAllContributorsInStorage(org) {
+    let contributorSet = new Set([]);
+    let myArr = [];
+    let repos = JSON.parse(localStorage.getItem('repos'));
+
+    return new Promise((resolve, reject) => {
+      repos.map((repo, i) => {
+        fetchRepoContributors(org, repo).then((repoContributors) => {
+          setTimeout(() => {
+            if (repoContributors.length > 0) {
+              repoContributors.map((contributor, i) => {
+              if(!contributorSet.has(contributor.login)) {
+                contributorSet.add(contributor.login);
+                myArr.push(contributor);
+              }
+            });
+            }
+            // Stores all unique contributors info to localstorage
+            localStorage.setItem('contributors', JSON.stringify(myArr));
```

Comment:
Interesting! When would this then get flushed and updated? Could we do once daily? And could we implement this via a `flushLocalStorageAllContributors` or something... so we could manually refresh it with a 🔄  button or something?

Reply:
Currently, the only way to request the API again for contributors is by manually flushing localStorage...but this can easily done, so no worries :)